docs(externals): add more explanation to externals (#185)
americanexpress/one-app-cli: packages/one-app-bundler/README.md

`providedExternals`, which is an array of external dependencies to be bundled
with it and provided to other modules.

> Modules shouldn't configure both `providedExternals` and `requiredExternals`.

Reviewer: ```suggestion
Modules shouldn't configure both `providedExternals` and `requiredExternals`.
Remember `providedExternals` are dependencies which your root module will make available to child modules. `requiredExternals` are a list of dependencies the child module will need to be made available by the root module.
```

> Any module with `requiredExternals` configured will be validated at runtime to ensure that
> the root module is in fact providing those `requiredExternals`, and will fail to load if it is
> not. For example, if your child module requires `^2.1.0` of a dependency but your root module provides `2.0.0`, One App will fail to load your child module as it does not satisfy the required semantic range.

Reviewer: ```suggestion
All modules `requiredExternals` are validated at runtime against the root modules list of `providedExternals`.  If the external dependency is not provided One App will throw an error. This will either result in the One App server not starting or, if it is already running, One App will not load that module. For example, if your child module requires `^2.1.0` of a dependency but your root module provides `2.0.0`, this will result in One App not loading that child module as the provided dependencies version does not satisfy the required semantic range.
This ensures that all of the listed dependencies features potentially required by the child module to work will be provided which could result in hard to debug bugs.
```

> Modules shouldn't configure both `providedExternals` and `requiredExternals`.

Reviewer 2: Would it be clearer to discuss this from the start of the section? Looking over the section on externals, it doesn't become apparent what each is used for (root and child module) until the end of the article (scattered between the lines).

Any module with `requiredExternals` configured will be validated at runtime to ensure that

> the root module is in fact providing those requiredExternals, and will fail to load if it is

Reviewer 2: ```suggestion
the root module is in fact providing those `requiredExternals`, and will fail to load if it is
```

> not. For example, if your child module is requiring `^2.1.0` of a dependency but your root module is providing `^2.0.0` of the dependency, One App will fail to load your child module as it is requiring a version that your root module is not providing.

Reviewer: ```suggestion
not. For example, if your child module requires `^2.1.0` of a dependency but your root module provides `2.0.0`, One App will fail to load your child module as it does not satisfy the required semantic range.
```

I also think more is needed here.
